### DISCUSSION TITLE
Is thumbnail cause pocketbase to freeze normal?

### DISCUSSION THREAD

**Discussion by joshuaavalon** [pocketbase/pocketbase#3637] [Q&A]
I create records with images via script. When I visit admin panel, it causes pocketbase to generate a lot a thumbnail at once and become not responsive unless the thumbnail is done generating. Is this a normal behavior?

I am using v0.19.1 and S3 storage.

**Comment by ganigeorgiev** [comment]
Thumbs are generated lazily only on the first access.

I guess it is possible the Admin UI to takes slightly longer if the first time it has to generate and load several thumbs at once. Using an S3 storage also can additionally contribute to the slow response times because each generated thumb will have to be uploaded to the remote S3 storage.

If the issue persists and the Admin UI is not responsive even after the first load, please provide more details about your setup (number of images per record, are you testing locally, is the S3 server operating within the same region as the PocketBase instance, etc.) so that I can try to reproduce it.

**Reply by ganigeorgiev** [reply]
_Another thing that could also affect the Admin UI thumbs rendering performance could be [the thumb component](https://github.com/pocketbase/pocketbase/blob/ea5ca009de839673d569da3d5950f8fcd4c59253/ui/src/components/records/RecordFileThumb.svelte) due to its JS state as it is not very optimized at the moment. I've added in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1) to profile it and see if can reduce its scripting time for the next release._

**Comment by joshuaavalon** [comment]
For your reference, I am using hosting it in  SGP1 of Digital Ocean with Cloudflare R2 as S3. I uploaded 10 records with 5 images each on 1 vCPU 1GB VM.

My guess of the problems are admin UI call all the images thumbnail at the same time cause all images to resize at the same time and use up all the CPU. 

I was excepting the image resizing to resize in queue and there is still CPU to response to other query. Also, if the image always shown in the admin UI, may be there should be an options to generate the image size for admin UI on upload (Perhaps an API argument to generate on upload).

**Reply by ganigeorgiev** [reply]
> I was excepting the image resizing to resize in queue and there is still CPU to response to other query

I'm not sure that I understand what do you mean by _"still CPU to response"_. We don't interact on low level with the CPU to have control over how much of it is being used and what exactly is being executed.
All requests are handled concurrently in their own individual goroutine and when and where they will be processed is managed by the Go runtime.

It is difficult to say what exactly is the main cause in your case - it could be network latency caused by the different servers location of DO and Cloudflare, but it could be also purely a client-side issue caused by a blocking scripting execution (_if you are still experiencing Admin UI unresponsiveness after the first thumbs generation, then most likely it is a client-side issue_) or it could be combination of both.

I'll try to investigate it in the near future and will make an attempt to optimize it.
